iceoryx_posh/test/moduletests/test_popo_client_port.cpp

```diff
@@ -0,0 +1,552 @@
+
+    static constexpr iox::units::Duration DEADLOCK_TIMEOUT{5_s};
+    Watchdog m_deadlockWatchdog{DEADLOCK_TIMEOUT};
+
+  public:
+    ClientPort_test()
+    {
+        iox::mepoo::MePooConfig mempoolconf;
+        mempoolconf.addMemPool({CHUNK_SIZE, NUM_CHUNKS});
+        m_memoryManager.configureMemoryManager(mempoolconf, m_memoryAllocator, m_memoryAllocator);
+    }
+
+    void SetUp() override
+    {
+        m_deadlockWatchdog.watchAndActOnFailure([] { std::terminate(); });
+
+        // this is basically what RouDi does when a client is requested
+        tryAdvanceToState(clientPortDataWithConnectOnCreate, iox::ConnectionState::CONNECTED);
+        tryAdvanceToState(clientPortDataWithoutConnectOnCreate, iox::ConnectionState::NOT_CONNECTED);
+    }
+
+    void TearDown() override
+    {
+    }
+
+    void tryAdvanceToState(ClientPortData& clientPortData, const iox::ConnectionState state)
+    {
+        ClientPortRouDi clientPortRouDi{clientPortData};
+        auto maybeCaProMessage = clientPortRouDi.tryGetCaProMessage();
+        if (state == iox::ConnectionState::NOT_CONNECTED && clientPortData.m_connectionState == state)
+        {
+            return;
+        }
+
+        ASSERT_TRUE(maybeCaProMessage.has_value());
+        auto& clientMessage = maybeCaProMessage.value();
+        ASSERT_THAT(clientMessage.m_type, Eq(iox::capro::CaproMessageType::CONNECT));
+        ASSERT_THAT(clientMessage.m_chunkQueueData, Ne(nullptr));
+        ASSERT_THAT(clientPortData.m_connectionState, Eq(iox::ConnectionState::CONNECT_REQUESTED));
+        if (clientPortData.m_connectionState == state)
+        {
+            return;
+        }
+
+        iox::capro::CaproMessage serverMessage{
+            iox::capro::CaproMessageType::ACK, m_serviceDescription, iox::capro::CaproMessageSubType::NOSUBTYPE};
+        serverMessage.m_chunkQueueData = &serverChunkQueueData;
+        clientPortRouDi.dispatchCaProMessageAndGetPossibleResponse(serverMessage);
+        ASSERT_THAT(clientPortData.m_connectionState, Eq(iox::ConnectionState::CONNECTED));
+        if (clientPortData.m_connectionState == state)
+        {
+            return;
+        }
+
+        constexpr bool NOT_IMPLEMENTED{true};
+        ASSERT_FALSE(NOT_IMPLEMENTED);
+    }
+
+    uint32_t getNumberOfUsedChunks() const
+    {
+        return m_memoryManager.getMemPoolInfo(0U).m_usedChunks;
+    }
+
+    iox::mepoo::SharedChunk getChunkFromMemoryManager(uint32_t userPayloadSize, uint32_t userHeaderSize)
+    {
+        return m_memoryManager.getChunk(iox::mepoo::ChunkSettings::create(userPayloadSize,
+                                                                          iox::CHUNK_DEFAULT_USER_PAYLOAD_ALIGNMENT,
+                                                                          userHeaderSize,
+                                                                          iox::CHUNK_DEFAULT_USER_PAYLOAD_ALIGNMENT)
+                                            .value());
```

Comment:
`ASSERT_FALSE` only works for functions returning `void`. I can do an `EXPECT_FALSE` but in this case I still have to return something and then this would need to be checked in every test case. Since the `MemoryManager` is not the component to test here, I took the pragmatic way. I could do a `cxx::Expects` to have more context if this really breaks one day but that one will most probably also be optimized away in release builds one day.

Reply:
Added a `cxx::Expects`. Please resolve if this is okay for you